### PR TITLE
MBS-10881 update to gradle 7.2

### PR DESCRIPTION
**Pull request by dsvoronin**
https://docs.gradle.org/7.2-rc-2/release-notes.html

also update detekt to 1.18.0, because of kotlin 1.5 dependency
disable "ArgumentListWrapping" rule, different behavior
fix "ImportOrdering" rule

### CODE REVIEW

**Comment by dsvoronin** [comment]
Waiting for release version

File: samples/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-2-all.zip
```
Comment (RuslanMingaliev) [inline samples/gradle/wrapper/gradle-wrapper.properties]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-3-all.zip
```

File: subprojects/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-2-all.zip
```
Comment (RuslanMingaliev) [inline subprojects/gradle/wrapper/gradle-wrapper.properties]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-rc-3-all.zip
```

File: subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt [hunk 3]
```diff
@@ -106,7 +106,8 @@ abstract class InHouseInstrumentationTestRunner :
         )
     }
 
-    @Suppress("MemberVisibilityCanBePrivate") // Public for *TestApp to skip on orchestrator runs
+    // Public for *TestApp to skip on orchestrator runs
+    @Suppress("MemberVisibilityCanBePrivate")
```
Comment (RuslanMingaliev) [inline subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/InHouseInstrumentationTestRunner.kt:110]: Already fixed here
https://github.com/pinterest/ktlint/pull/1168

File: build-logic/build.gradle.kts [hunk 4]
```diff
@@ -1,11 +1,13 @@
+import io.gitlab.arturbosch.detekt.Detekt
+
 plugins {
     base
-    // accessing version catalog here will be supported in grale 7.2
-    // https://github.com/gradle/gradle/pull/17394
-    id("io.gitlab.arturbosch.detekt") version "1.16.0"
+    // accessing version catalog here is blocked by IDE false-postive error
+    // https://youtrack.jetbrains.com/issue/KTIJ-19369
+    id("io.gitlab.arturbosch.detekt") version "1.18.0"
```
Comment (RuslanMingaliev) [inline build-logic/build.gradle.kts:7]: Remove the `jetbrains-kotlin-html` repo

File: conf/detekt.yml [hunk 5]
```diff
@@ -262,8 +262,9 @@ formatting:
         active: false
     AnnotationSpacing:
         active: true
+    # todo fix and enable
     ArgumentListWrapping:
```
Comment (RuslanMingaliev) [inline conf/detekt.yml:266]: https://github.com/pinterest/ktlint/pull/1196
This is fixed too

Comment (RuslanMingaliev) [reply conf/detekt.yml:266]: That's why I didn't update the Detekt version [here](https://github.com/avito-tech/avito-android/pull/1163) :)

Comment (dsvoronin) [reply conf/detekt.yml:266]: We need to run ktlint separately i think
MBS-11763

File: build-logic/build.gradle.kts (hunk 4 above)
Comment (dsvoronin) [reply build-logic/build.gradle.kts:7]: done

File: samples/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (dsvoronin) [reply samples/gradle/wrapper/gradle-wrapper.properties]: fixed, thanks

File: subprojects/gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (dsvoronin) [reply subprojects/gradle/wrapper/gradle-wrapper.properties]: fixed

**Comment by dsvoronin** [comment]
I think it's safe to merge now;
will update to stable version when it's available

**Comment by RuslanMingaliev** [comment]
Close also MBS-11267

**Review by RuslanMingaliev — APPROVED** [review]

**Pull request merged**